demo/rust/micdemo/src/main.rs

```diff
@@ -0,0 +1,299 @@
+/*
+    Copyright 2018-2021 Picovoice Inc.
+
+    You may not use this file except in compliance with the license. A copy of the license is located in the "LICENSE"
+    file accompanying this source.
+
+    Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+    an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+    specific language governing permissions and limitations under the License.
+*/
+
+use clap::{App, Arg, ArgGroup};
+use ctrlc;
+use miniaudio;
+use pv_porcupine::{pv_keyword_paths, pv_library_path, pv_model_path, Porcupine, KEYWORDS};
+use std::path::PathBuf;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::time::SystemTime;
+
+static LISTENING: AtomicBool = AtomicBool::new(true);
```

Comment:
a bit pedantic, but should probably set this to false first, then when you're ready to launch the recording loop, set to true, then set to false again